Compiler crash caused by codegen from concepts

```
type
  Thing = concept x
    x.hello is string
  Cat = object

proc hello(d: Cat): string = "Meow"

proc sayHello(c: Thing) = echo(c.hello)

var a: Thing = Cat()
a.sayHello()
```


This crashes on compilation with the following error:

> Error: execution of an external compiler program 'clang -c  -w  -I/usr/local/Cellar/nim/0.17.2/nim/lib -o /Users/rbreen/Dev/nim/test/nimcache/concepts.o /Users/rbreen/Dev/nim/test/nimcache/concepts.c' failed with exit code: 1
>
>/Users/rbreen/Dev/nim/test/nimcache/concepts.c:145:8: error: used type 'tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q' (aka 'struct tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q') where arithmetic or pointer type is required
        T1_ = (tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q)0;
              ^                                    ~
>/Users/rbreen/Dev/nim/test/nimcache/concepts.c:146:8: error: used type 'tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q' (aka 'struct tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q') where arithmetic or pointer type is required
        T1_ = (tyObject_Cat_cxAS31s7gVkqKcpYVzkF3Q)0;

It compiles correctly if I change one line to:

```
var a: Cat = Cat()
```

This was tested in Nim 0.17.2.

--- a/compiler/cgen.nim
+++ b/compiler/cgen.nim
@@ -288,7 +288,7 @@ type
 proc genRefAssign(p: BProc, dest, src: TLoc, flags: TAssignmentFlags)
 
 proc isComplexValueType(t: PType): bool {.inline.} =
-  let t = t.skipTypes(abstractInst)
+  let t = t.skipTypes(abstractInst + tyUserTypeClasses)
   result = t.kind in {tyArray, tySet, tyTuple, tyObject} or
     (t.kind == tyProc and t.callConv == ccClosure)
 

--- a/tests/concepts/tmisc_issues.nim
+++ b/tests/concepts/tmisc_issues.nim
@@ -9,7 +9,8 @@ implicit generic
 generic
 false
 true
--1'''
+-1
+Meow'''
 """
 
 # https://github.com/nim-lang/Nim/issues/1147
@@ -98,3 +99,15 @@ let b = B()
 echo b is A
 echo b.size()
 
+# https://github.com/nim-lang/Nim/issues/7125
+type
+  Thing = concept x
+    x.hello is string
+  Cat = object
+
+proc hello(d: Cat): string = "Meow"
+
+proc sayHello(c: Thing) = echo(c.hello)
+
+var a: Thing = Cat()
+a.sayHello()

